scapy/utils.py

```diff
@@ -491,14 +491,16 @@ def do_graph(graph,prog=None,format=None,target=None,type=None,string=None,optio
             target = get_temp_file(autoext="."+format)
             start_viewer = True
         else:
-            target = "| %s" % conf.prog.display
+            with ContextManagerSubprocess("do_graph()"):
+                target = subprocess.Popen([conf.prog.display], stdin=subprocess.PIPE).stdin
     if format is not None:
         format = "-T%s" % format
     if isinstance(target, str):
         target = open(os.path.abspath(target), "wb")
-    proc = subprocess.Popen("%s %s %s" % (prog, options or "", format or ""),
+    proc = subprocess.Popen("\"%s\" %s %s" % (prog, options or "", format or ""),
```

Comment:
explanation: when there is a space in the path, it crashes (e.g. windows: "Program Files")